### PR TITLE
feat(dli): add new datasource to get list of sql templates

### PR DESCRIPTION
**Pull request by wangyuancheng01**
**What this PR does / why we need it**:
   add new datasource to get list of sql templates

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/dli" TESTARGS="-run TestAccDataSourceDliSqlTemplates_basic"
...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run TestAccDataSourceDliSqlTemplates_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceDliSqlTemplates_basic
=== PAUSE TestAccDataSourceDliSqlTemplates_basic
=== CONT  TestAccDataSourceDliSqlTemplates_basic
--- PASS: TestAccDataSourceDliSqlTemplates_basic (46.91s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       46.957s
```

### CODE REVIEW

File: docs/data-sources/dli_sql_templates.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  The [templates](#templates_struct) structure is documented below.
+
+<a name="templates_struct"></a>
+The `templates` block supports:
+
+* `id` - The ID of SQL template.
+
+* `name` - The name of SQL template.
+
+* `sql` - The text of SQL template.
```
Comment (Lance52259) [inline docs/data-sources/dli_sql_templates.md]:
```suggestion
* `sql` - The SQL statement of SQL template.
```

Comment (wangyuancheng01) [reply docs/data-sources/dli_sql_templates.md]: done

**Pull request merged**